### PR TITLE
Added sanity pipeline for PG migration with complex schemas

### PR DESCRIPTION
**Pull request by shubham-yb**
- Added a new script for schema migration with validations at each step and file replacement provisions.
- Added 9 complex schemas to the PG tests which will serve as a sanity run.
- Added a few more functions in the YB lib for better validations.

### CODE REVIEW

**Comment by shubham-yb** [comment]
https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline-test/681
https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline-test/680

**Comment by priyanshi-yb** [comment]
@shubham-yb whenever you get a chance, please update the expected files of the schemas in the PR as per the latest main.

**Review by priyanshi-yb — APPROVED** [review]
LGTM

File: migtests/tests/pg/omnibus/sql/00_create_users_roles.ddl.sql [hunk 1]
```diff
@@ -0,0 +1,28 @@
+-- running with role postgres with admin privileges
```
Comment (priyanshi-yb) [inline migtests/tests/pg/omnibus/sql/00_create_users_roles.ddl.sql:1]: General comment: For the Schema, see if we can use the zipped file and unzip it in the script to reduce the size/number of files in the repo

File: migtests/tests/pg/mgi/replacement_dir/tables/INDEXES_table.sql [hunk 2]
```diff
@@ -0,0 +1,2255 @@
...
+SET idle_in_transaction_session_timeout = 0;
+SET client_encoding = 'UTF8';
+SET standard_conforming_strings = on;
+SELECT pg_catalog.set_config('search_path', '', false);
+SET xmloption = content;
+SET client_min_messages = warning;
+SET row_security = off;
+
+
+CREATE INDEX acc_accession_0 ON mgd.acc_accession USING btree (lower(accid));
```
[the remark this answers is not in the document]
Comment (shubham-yb) [reply migtests/tests/pg/mgi/replacement_dir/tables/INDEXES_table.sql:13]: There are a couple of ALTER TABLES in INDEXES_table.sql.

```/*
ERROR: ALTER TABLE CLUSTER not supported yet (SQLSTATE 0A000)
File :/home/ubuntu/yb-voyager/migtests/tests/pg/mgi/export-dir/schema/tables/INDEXES_table.sql
*/
ALTER TABLE mgd.all_cre_cache CLUSTER ON all_cre_cache_idx_clustered;

/*
ERROR: ALTER TABLE CLUSTER not supported yet (SQLSTATE 0A000)
File :/home/ubuntu/yb-voyager/migtests/tests/pg/mgi/export-dir/schema/tables/INDEXES_table.sql
*/
ALTER TABLE mgd.all_knockout_cache CLUSTER ON all_knockout_cache_idx_clustered;
```

These should've been classified in table.sql itself. Not sure if its a bug.

**Comment by shubham-yb** [comment]
https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline-test/733/

File: migtests/tests/pg/mgi/replacement_dir/tables/INDEXES_table.sql (hunk 2 above)
[the remark this answers is not in the document]
Comment (priyanshi-yb) [reply migtests/tests/pg/mgi/replacement_dir/tables/INDEXES_table.sql:13]: Ohh, that might be something to understand if there is anything special for these `CLUSTER ON` DDLs.

File: migtests/tests/pg/rna/expected_files/expectedAssessmentReport.json
Comment (priyanshi-yb) [inline migtests/tests/pg/rna/expected_files/expectedAssessmentReport.json:1]: See if it is easier to zip the expected files as it is large for of these.

**Pull request merged**